### DISCUSSION TITLE
Criação de chatbot com Telegram

### DISCUSSION THREAD

**Discussion by LuisMatheus-dev** [frontendbr/forum#2060] [Perguntas e Respostas]
Olá, 
Eu estou com um projeto de um bot para o Telegram ,como nunca trabalhei com nada além de arquivos locais , essa ideia ficou muito abstrata para min, e gostaria de alguma sugestão sobre a implementação.

O core seria uma função, que saberia a frequência de pedidos de um cliente (a cada 2 semanas, 1 mês , 4 dias...), quando chegasse a esse intervalo, ele entraria em contato com o cliente diretamente pelo Telegram , dizendo se ele gostaria de efetuar o pedido, recebendo esse pedido (pdf ou mensagem de texto) ele enviaria o pedido formatado para meu email. 

Minha dúvida principal seria na estrutura do projeto, eu vou precisar colocar esses arquivos dentro de uma plataforma de nuvem ? de um banco de dados ? Tem algum detalhe importante que não posso deixar para trás ?

**Comment by lucaspontoexe** [comment]
Bom, você pode começar vendo sobre a API de bots que o Telegram disponibiliza
https://core.telegram.org/bots

Aqui tem alguns exemplos de códigos e bibliotecas pra várias linguagens:
https://core.telegram.org/bots/samples


>  eu vou precisar colocar esses arquivos dentro de uma plataforma de nuvem?

Sim, você vai precisar ter um back-end rodando em algum servidor pra processar as mensagens que o seu bot vai receber. E como você vai armazenar a quantidade de pedidos do cliente, pode precisar de um banco de dados também.
